src/main/io/vtx_tramp.c

```diff
@@ -317,175 +260,211 @@ static char trampReceive(uint32_t currentTimeUs)
                 }
             }
-
-            lastQueryTimeUs = currentTimeUs;
+            break;
         }
-        break;
-
-    case TRAMP_STATUS_SET_FREQ_PW:
+        case TRAMP_STATUS_ONLINE_MONITOR_TEMP:
         {
-            bool done = true;
-            if (trampConfFreq && trampFreqRetries && (trampConfFreq != trampCurFreq)) {
-                trampSendFreq(trampConfFreq);
-                trampFreqRetries--;
-#ifdef TRAMP_DEBUG
-                debugFreqReqCounter++;
-#endif
-                done = false;
-            } else if (trampConfPower && trampPowerRetries && (trampConfPower != trampConfiguredPower)) {
-                trampSendRFPower(trampConfPower);
-                trampPowerRetries--;
-#ifdef TRAMP_DEBUG
-                debugPowReqCounter++;
-#endif
-                done = false;
+            // Check request time
+            if (replyCode == 's') {
+                // Got reply, return to request freq/power/pit
+                trampStatus = TRAMP_STATUS_ONLINE_MONITOR_TEMP;
+            } else if (cmp32(currentTimeUs, trampLastTimeUs) >= TRAMP_MIN_REQUEST_PERIOD_US) {
+                // Timed out after min request period, return to request freq/power/pit query
+                trampStatus = TRAMP_STATUS_ONLINE_MONITOR_FREQPWRPIT;
             }
+            break;
+        }
+        case TRAMP_STATUS_ONLINE_CONFIG:
+        {
+            // Param should now be set, check time
+            if (cmp32(currentTimeUs, trampLastTimeUs) >= TRAMP_MIN_REQUEST_PERIOD_US) {
+                // Min request period exceeded, re-query
+                trampQuery('v');
+
+                // Advance state
+                trampStatus = TRAMP_STATUS_ONLINE_MONITOR_FREQPWRPIT;
 
-            if (!done) {
-                trampStatus = TRAMP_STATUS_CHECK_FREQ_PW;
-
-                // delay next status query by 300ms
-                lastQueryTimeUs = currentTimeUs + 300 * 1000;
-            } else {
-                // everything has been done, let's return to original state
-                trampStatus = TRAMP_STATUS_ONLINE;
-                // reset configuration value in case it failed (no more retries)
-                trampConfFreq  = trampCurFreq;
-                trampConfPower = trampConfiguredPower;
-                trampFreqRetries = trampPowerRetries = 0;
+                // Update last time
+                trampLastTimeUs = currentTimeUs;
             }
+            break;
         }
-        break;
-
-    case TRAMP_STATUS_CHECK_FREQ_PW:
-        if (cmp32(currentTimeUs, lastQueryTimeUs) > 200 * 1000) {
-            trampQueryV();
-            lastQueryTimeUs = currentTimeUs;
+        default:
+        {
+            // Invalid state, reset
+            trampStatus = TRAMP_STATUS_OFFLINE;
+            break;
         }
-        break;
-
-    default:
-        break;
     }
 
 #ifdef TRAMP_DEBUG
-    debug[1] = debugFreqReqCounter;
-    debug[2] = debugPowReqCounter;
+    debug[1] = trampStatus;
```

Comment:
Instead of using these compile-time debug settings (that conflict with runtime debug settings), this should be made into a proper debug mode and use `DEBUG_SET()`. See #9832 for an example.

Reply:
New debug option "VTX_TRAMP" added, replacing the compile time debug. Wish I'd known about that when I was working on it before. Checked that the debug can be enabled and viewed from within betaflight configurator.